### DISCUSSION TITLE
ITryGetValue Interface

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#73015] [Ideas]
In .NET 7.0, a new interface, ```IParsable``` was added that is awesome for standardizing all parsing interfaces.

There are a number of classes that  a ```TryGetValue``` method.  It would be great if these could be adjusted to implement an interface so that code can target the interface directly instead of all implementors of it.

There are two common shapes to TryGetValue:
```
public delegate bool TryGetValue<TResult>([NotNullWhen(true)] out TResult? Result);
public delegate bool TryGetValue<TInput, TResult>(TInput Input, [NotNullWhen(true)] out TResult? Result);
```

It would be great if an ITryGetValue<TResult> and an ITryGetValue<TInput, TResult> interface were introduced.

**Comment by bartonjs** [comment]
Successful interfaces describe capability, which we get from the `-able` suffix.

`IParsable` is an inverse capability to `IFormattable`; an object that is both of those can serialize itself to a string; and we have an actual use for it from the generic math project.

TryGetValue as an interface seems like it's just syntax to make sure everyone does it the same way... I don't know what real use it would have as a type (what would take an `ITryGetValue<T1,T2>` as a parameter?).  And "just syntax" is not something we'd add as an interface.
